### PR TITLE
UDF types

### PR DESCRIPTION
**Pull request by akshatsikarwar**
/silent

### CODE REVIEW

**Review by riverszhang89 — APPROVED** [review]
Looks good to merge.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0093c29/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/373 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>udf</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0093c29/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0093c29/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2504'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/373 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>udf</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/4c64b8d/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/4c64b8d/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2504'>Resubmit</a></p>

File: tests/udf.test/runit [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+cdb2sql --tabs --script ${1} ${CDB2_OPTIONS} - 2> actual 1>&2 <<'EOF'
```
Comment (riverszhang89) [inline tests/udf.test/runit:2]: Are you missing `default` here?

Comment (akshatsikarwar) [reply tests/udf.test/runit]: yeah it was bad copy pasta

Comment (nirbhayc) [reply tests/udf.test/runit]: You could instead source tests/tools/compare_results.sh to avoid this boilerplate.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/373 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/603b265/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/603b265/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2504'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/e809ed2/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/373 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>cldeadlock [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/e809ed2/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/e809ed2/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2504'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/e809ed2/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/373 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/e809ed2/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/e809ed2/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2504'>Resubmit</a></p>

**Review by roborivers — COMMENTED** [review]
<p>Sysbench Transactions Per Second (commit: e809ed2):</p>

<pre lang=diff>
@@ sysbench/select_random_points @@
failed
@@ sysbench/bulk_insert @@
+4.74%
@@ sysbench/select_random_ranges @@
+9.68%
@@ sysbench/oltp_insert @@
+4.03%
@@ sysbench/oltp_read_only @@
+11.26%
@@ sysbench/oltp_point_select @@
+9.57%
@@ sysbench/oltp_update_non_index @@
+7.46%
@@ sysbench/oltp_delete @@
+7.29%
@@ sysbench/oltp_update_index @@
+6.23%
@@ sysbench/oltp_read_write @@
+9.03%
@@ sysbench/oltp_write_only @@
+8.09%
</pre>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/373 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c756196/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c756196/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2504'>Resubmit</a></p>

**Pull request merged**